internal/common/common.go

```diff
@@ -150,8 +151,37 @@ func GetNetworkCidrAttr(obj interface{}, fieldName string) []*string {
 	}
 
 	funk.ForEach(field, func(elem interface{}) {
-		addresses = append(addresses, swag.String(string(funk.Get(elem, "Cidr").(models.Subnet))))
+		address := funk.Get(elem, "Cidr")
+		if address != nil {
+			addresses = append(addresses, swag.String(string(address.(models.Subnet))))
+		}
 	})
 
 	return addresses
 }
+
+// IsNetworksSliceNonEmpty checks whether the provided slice of networks, i.e. objects containing
+// CIDR is non-empty. The slice is assumed to be non-empty if at least one of its elements
+// contains a non-empty CIDR. Examples:
+// - `[]*models.MachineNetwork{{Cidr: "5.5.0.0/24"}, {Cidr: "6.6.0.0/24"}}` - valid, as we are
+//   configuring two machine networks
+// - `[]*models.ClusterNetwork{}` - valid, as it means we are removing all the cluster networks
+//   that may have been currently configured
+// - `[]*models.MachineNetwork{{}}` - invalid, as it means that we are trying to configure
+//    a single machine network that is empty; a valid network contains at least a CIDR which is
+//    missing in this case
+// - `[]*models.MachineNetwork{{Cidr: ""}}` - invalid, as it means we are trying to configure
+//   a single machine network that has empty CIDR; a valid network should contain a non-empty
+//   CIDR
+// - `[]*models.ClusterNetwork{{HostPrefix: 0}}` - invalid, as it means we are trying to configure
+//   a single cluster network with host prefix with a value 0; this is not a valid subnet lenght
+func IsNetworksSliceNonEmpty(arg interface{}) bool {
```

Comment:
I think this function could be simplified to be `IsSliceNonEmpty`
With going over all the fields of the elements and checking in case they are a pointer, validate that they are not nil
pseudo:
```golang
v := reflect.ValueOf(arg)
for i := 0; i < v.NumField(); i++ {
    res = res || v.Kind() == reflect.Ptr && reflectv.Field(i).IsNil()
}

Reply:
Done, it covers a generic case now